### PR TITLE
[BEAM-2436] table is not regeisted in BeamSql.query

### PR DESCRIPTION
**Pull request by XuMingmin**
R: @takidau @xumingming 
CC: @lukecwik

### CODE REVIEW

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java [hunk 1]
```diff
@@ -58,16 +60,31 @@ public static void main(String[] args) throws Exception {
     PCollection<BeamSQLRow> inputTable = PBegin.in(p).apply(Create.of(row)
         .withCoder(new BeamSqlRowCoder(type)));
 
-    //run a simple SQL query over input PCollection;
-    String sql = "select c2, c3 from TABLE_A where c1=1";
-    PCollection<BeamSQLRow> outputStream = inputTable.apply(BeamSql.simpleQuery(sql));
+    //Case 1. run a simple SQL query over input PCollection with BeamSql.simpleQuery;
+    PCollection<BeamSQLRow> outputStream = inputTable.apply(
+        BeamSql.simpleQuery("select c2, c3 from TABLE_A where c1=1"));
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: The usage of `BeamSql.simpleQuery` seems a little weird: user have never specified the table name: `TABLE_A`.

Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: `simpleQuery` is added for single table SQL. As it's not possible to name a `PCollection` here, it's automately set to table name used in query.

This brings a potential issue of table name conflict, the solution may be limiting the scope of table schema, will open a new task to talk.

Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: Ok, let's talk further in the new task.

**Comment by xumingming** [comment]
LGTM

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java (hunk 1 above)
Comment (takidau) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: +1 to James' concerns.

**Comment by XuMingmin** [comment]
rebase to fix conflict.

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java (hunk 1 above)
Comment (lukecwik) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: Its extremely convenient to not have to name your tables in the simpleQuery approach.

Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: @takidau as talking about the design of interface, the default method is `public static PTransform<PCollectionTuple, PCollection<BeamSqlRow>> query(String sqlQuery)`, which relies on the named `TupleTag` to specify table name. 

`public static PTransform<PCollection<BeamSqlRow>, PCollection<BeamSqlRow>>
  simpleQuery(String sqlQuery)` is a special case which runs on single table/PCollection. There's no exiting method to name it, so the table name in query is take as granted.

For both methods, potentially the table name would be mixed up, that's why I said a further task is needed to have a separated schema namespace for each query. 

@lukecwik  any comments?

Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: @lukecwik do you mean `pcollection.simpleQuery('SELECT C1, C2')`, to replace `pcollection.simpleQuery('SELECT C1, C2 FROM TABLE_NAME')`?

Comment (lukecwik) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: @XuMingmin Having the ability to use `'SELECT C1, C2 FROM TABLE_NAME'` or `'SELECT C1, C2'` doesn't matter to me. What matters to me is that they don't need to use a `PCollectionTuple` with a single `PCollection` and `TupleTag` pair.

I was thinking that the whole idea of global registration whenever someone calls `query` or `simpleQuery` will impact the users future pipeline construction depending on the order in which they apply parts of their pipeline is not a good idea. I can see how its useful in a CLI where they aren't building a pipeline programmatically but should be limited to paths which the CLI code handles.

Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java]: Agree, a global schema namespace doesn't sound good in DSL, would address the issue in a new task.

**Comment by XuMingmin** [comment]
@lukecwik @takidau any pending requests here? 
If not, could you help to merge it?

**Comment by lukecwik** [comment]
I don't think you want to register tables with BeamSqlEnv within expand(). BeamSqlEnv is currently the global namespace.

**Comment by XuMingmin** [comment]
BeamSqlEnv has an empty schema repository. As we don't declare the table explicitly with `BeamSql.registerTable(tableName, PCollection<BeamSqlRow>)`, this is the row to do the job.

**Comment by lukecwik** [comment]
SGTM
Merging.

**Comment by XuMingmin** [comment]
Thanks @lukecwik, @xumingming 

Created BEAM2446 to track the next step.

**Pull request closed without merging**